### DISCUSSION TITLE
Handling of batched processing which may contain invalid video ids.

### DISCUSSION THREAD

**Discussion by 2meito** [sigma67/ytmusicapi#435] [Q&A]
Hello, I'm currently using this library with browser authentication to create a program that can read a text file containing video ids and add them to a playlist.

However, some of those video ids are from videos are either private or no longer exists, so when I try to run:

```python
response = ytmusic.add_playlist_items(playlist_id, video_ids)
```

I usually get a response like this:
```bash
Server returned HTTP 400: Bad Request.
Precondition check failed.
```

Currently I'm trying to see how far I can go while sending individual request, but I was wondering if there was a better way to handle batched process for a problem like this.

**Comment by sigma67** [comment]
This is an interesting question, you're basically looking for a way to validate the videoId before calling the function proper.

The only way I can think of doing this would be to all `get_song` or `get_watch_playlist` on each `videoId` and see what you get. I'd expect one of those to throw an error as well, or at least return different output than an available song.
